core/lib/Cardano/Address/Style/Shared.hs

```diff
@@ -0,0 +1,323 @@
+            \number of addresses for a given wallet."
+
+-- | Generate a root key from a corresponding mnemonic.
+--
+-- @since 3.4.0
+genMasterKeyFromMnemonic
+    :: SomeMnemonic
+        -- ^ Some valid mnemonic sentence.
+    -> ScrubbedBytes
+        -- ^ An optional second-factor passphrase (or 'mempty')
+    -> Shared 'RootK XPrv
+genMasterKeyFromMnemonic =
+    Internal.genMasterKeyFromMnemonic
+
+-- | Generate a root key from a corresponding root 'XPrv'
+--
+-- @since 3.4.0
+genMasterKeyFromXPrv
+    :: XPrv -> Shared 'RootK XPrv
+genMasterKeyFromXPrv =
+    Internal.genMasterKeyFromXPrv
+
+-- Re-export from 'Cardano.Address.Derivation' to have it documented specialized in Haddock.
+--
+-- | Derives an account private key from the given root private key.
+--
+-- @since 3.4.0
+deriveAccountPrivateKey
+    :: Shared 'RootK XPrv
+    -> Index 'Hardened 'AccountK
+    -> Shared 'AccountK XPrv
+deriveAccountPrivateKey =
+    Internal.deriveAccountPrivateKey
+
+-- Re-export from 'Cardano.Address.Derivation' to have it documented specialized in Haddock.
+--
+-- | Derives a multisig private key from the given account private key for payment credential.
+--
+-- @since 3.4.0
+deriveMultisigForPaymentPrivateKey
+    :: Shared 'AccountK XPrv
+    -> Index 'Soft 'PaymentK
+    -> Shared 'ScriptK XPrv
+deriveMultisigForPaymentPrivateKey accPrv addrIx =
+    let (Shared xprv) = Internal.deriveAddressPrivateKey accPrv UTxOExternal addrIx
+    in Shared xprv
+
+-- Re-export from 'Cardano.Address.Derivation' to have it documented specialized in Haddock.
+--
+-- | Derives a multisig private key from the given account private key for delegation credential.
+--
+-- @since 3.4.0
+deriveMultisigForDelegationPrivateKey
+    :: Shared 'AccountK XPrv
+    -> Index 'Soft 'PaymentK
+    -> Shared 'ScriptK XPrv
+deriveMultisigForDelegationPrivateKey accPrv addrIx =
+    let (Shared xprv) = Internal.deriveAddressPrivateKey accPrv Stake addrIx
+    in Shared xprv
+
+-- Re-export from 'Cardano.Address.Derivation' to have it documented specialized in Haddock
+--
+-- | Derives a multisig public key from the given account public key for payment credential.
+--
+-- @since 3.4.0
+deriveMultisigForPaymentPublicKey
+    :: Shared 'AccountK XPub
+    -> Index 'Soft 'PaymentK
+    -> Shared 'ScriptK XPub
+deriveMultisigForPaymentPublicKey accPub addrIx =
+    let (Shared xpub) = Internal.deriveAddressPublicKey accPub UTxOExternal addrIx
+    in Shared xpub
+
+-- Re-export from 'Cardano.Address.Derivation' to have it documented specialized in Haddock
+--
+-- | Derives a multisig public key from the given account public key for delegation credential.
+--
+-- @since 3.4.0
+deriveMultisigForDelegationPublicKey
+    :: Shared 'AccountK XPub
+    -> Index 'Soft 'PaymentK
+    -> Shared 'ScriptK XPub
+deriveMultisigForDelegationPublicKey accPub addrIx =
+    let (Shared xpub) = Internal.deriveAddressPublicKey accPub Stake addrIx
+    in Shared xpub
+
```

Comment:
We could stick with:

```
    , deriveAccountPrivateKey
    , deriveAddressPrivateKey
    , deriveDelegationPrivateKey
    , deriveAddressPublicKey
```

For all these to be consistent with Shelle and Icarus wallets. In the end, the "multisig" aspect is implied by the Shared key already.

Reply:
makes sense. will do it